src/util/RangeUtil.v3

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020 Ben L. Titzer. All rights reserved.
+// See LICENSE for details of Apache 2.0 license.
+
+// Utility methods for vector.
+// 
+// TODO: all these searches can be done on Range<T>. Maybe add a `toRange`
+// method to `Vector` in virgil and change this to `RangeSearchUtil`? Vector<T>
+// to Range<T> should be pretty much free.
+component RangeUtil {
+  // Assumes that the vector is sorted.
+  def binarySearch<T>(range: Range<T>, val: T, lt: (T, T) -> bool) -> int {
```

Comment:
I like this method. I think it should live in Virgil's `lib/util` and have some unittests.